wallet/wallet.go

```diff
@@ -3307,7 +3307,7 @@ func (w *Wallet) PublishTransaction(tx *wire.MsgTx) error {
 // from the database (along with cleaning up all inputs used, and outputs
 // created) if the transaction is rejected by the back end.
 func (w *Wallet) publishTransaction(tx *wire.MsgTx) (*chainhash.Hash, error) {
-	server, err := w.requireChainClient()
+	chainClient, err := w.requireChainClient()
```

Comment:
https://github.com/btcsuite/btcwallet/commit/e508a127b6eb0c07dc7d9e6380a08eb0c95df7ae

Reply:
Looking at that commit, it looks like it only touched notifications when sending to our own addresses though?

Reply:
Yeah, that's what the issue is related to. The method was refactored to allow the backend interaction to happen outside of the database transaction, but we didn't account for this when generating our own change outputs within `SendOutputs`.

Reply:
Ah, right. So if I understand this correctly:
* We previously would ask the backend to notify us every time we called `new(Change)Address`, within the DB transaction.
* The mentioned commit moved this `NotifyReceived` call outside the `new(Change)Address` method, making the caller of the method responsible for calling the backend.
* This was correctly done in `New(Change)Address`, but not in `SendOutputs`, where `newChangeAddress` is called directly.

Reply:
With the above in mind, even though this surely addresses the current problem, I'm not sure if this is the _correct_ way of fixing it. 

For instance is it still possible for `CreateSimpleTx` to be called, crafting a TX with a new change address, but we are not guaranteed it will be published through `publishTransaction`.

Instead we could require the caller of `newChangeAddress` to **always** register the address with the backend. In this case that would entail making sure the address from the `changeSource` in `txToOutputs` is always registered. If you want to do this outside the db transaction, I imagine `OnCommit` can be used.